src/model/SequenceView.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import { InfoList, Sequence } from 'franklin-sites';
+import { FreeTextType } from './FreeText';
+import apiUrls from '../utils/apiUrls';
+import fetchData from '../utils/fetchData';
+import { formatLargeNumber } from '../utils/utils';
+import idx from 'idx';
+
+enum Flag {
+  PRECURSOR = 'Precursor',
+  FRAGMENT = 'Fragment',
+  FRAGMENTS = 'Fragments',
+  FRAGMENT_PRECURSOR = 'Fragment,Precursor',
+  FRAGMENTS_PRECURSOR = 'Fragments,Precursor',
+}
+
+type Isoform = {
+  name: { value: string };
+  isoformSequenceStatus: string;
+  isoformIds: string[];
+  synonyms: Array<{ value: string }>;
+  note: { texts: Array<{ value: string }> };
+  sequenceIds: string[];
+};
+
+type AlternativeProduct = {
+  commentType: FreeTextType.ALTERNATIVE_PRODUCTS;
+  isoforms: Isoform[];
+  note: { texts: Array<{ value: string }> };
+  events: string[];
+};
+
+type SequenceViewProps = {
+  data: {
+    entryAudit?: {
+      lastSequenceUpdateDate?: string;
+      sequenceVersion: string;
+    };
+    proteinDescription?: {
+      flag?: Flag;
+    };
+    sequence?: {
+      value: string;
+      length: number;
+      molWeight: number;
+      crc64: string;
+    };
+    comments?: AlternativeProduct[];
+  };
+};
+
+export const SequenceInfo: React.FC<{
+  isoformId: string;
+  isoformSequence?: SequenceViewProps['data']['sequence'];
+  lastUpdateDate?: string | null;
+}> = ({ isoformId, isoformSequence, lastUpdateDate }) => {
+  const [data, setData] = useState(null);
+  const [isoformToFetch, setIsoformToFetch] = useState('');
+
+  useEffect(() => {
+    if (!isoformToFetch) {
+      return;
+    }
+    const fetchIsoformData = async () => {
+      const result = await fetchData(`${apiUrls.entry(isoformToFetch)}`);
+      setData(result.data.sequence);
```

Comment:
Would check for existence of the path `result.data.sequence` before accessing it.